### PR TITLE
rbac: polish error caused by missing the permission to register graph applications

### PR DESCRIPTION
**Pull request by yugangw-msft**
Per feedback from regional directors and MVP

- [na] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by yugangw-msft** [comment]
@williexu, could you please take a look? Looks like github requires an approval from a reviewer with write access

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -970,7 +971,15 @@ def create_application(client, display_name, homepage, identifier_uris,
...
-    return client.create(app_create_param)
+    try:
+        return client.create(app_create_param)
+    except GraphErrorException as ex:
+        if 'insufficient privileges' in str(ex).lower():
+            link = 'https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal'  # pylint: disable=line-too-long
+            raise CLIError("Directory permission is needed for the current user to register the application. "
+                           "For how to configure, please refer '{}'. Original error: {}".format(link, ex))
+        else:
+            raise
```
Comment (williexu) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: No need for the `else:`, though I guess it doesn't hurt.

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/tests/test_role_commands_thru_mock.py [hunk 2]
```diff
@@ -157,10 +157,52 @@ def mock_app_create(parameters):
...
+        def _test_deserializer(resp_type, response):
+            err = FakedError('something bad for you')
+            return err
+
+        faked_role_client = mock.MagicMock()
+        faked_role_client.config.subscription_id = self.subscription_id
+        auth_client_mock.return_value = faked_role_client
+        faked_graph_client = mock.MagicMock()
+        graph_client_mock.return_value = faked_graph_client
+
```
Comment (williexu) [inline src/command_modules/azure-cli-role/azure/cli/command_modules/role/tests/test_role_commands_thru_mock.py]: It looks like there's some duplicate code in these two tests, like the `_test_deserializer` code.

**Comment by yugangw-msft** [comment]
> Some comments, also, why isn't the HISTORY for azure-cli-acs being updated?

Because I forgot it

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/tests/test_role_commands_thru_mock.py (hunk 2 above)
Comment (yugangw-msft) [reply src/command_modules/azure-cli-role/azure/cli/command_modules/role/tests/test_role_commands_thru_mock.py]: Indeed, let me get rid of

**Comment by yugangw-msft** [comment]
@williexu, can you take another look?

**Pull request merged**